pkg/metadata/storage/registry.go

```diff
@@ -107,44 +114,67 @@ func (r *registryBackend) WriteObject(ctx context.Context, fpath string, obj int
 	}
 
 	// Write metadata to disk for packing into archive
-	if err := r.localDirBackend.WriteObject(ctx, fpath, obj); err != nil {
+	if err := b.localDirBackend.WriteObject(ctx, fpath, obj); err != nil {
 		return err
 	}
-	logrus.Debugf("Pushing metadata to registry at %s", r.src)
-	return r.pushImage(data, fpath)
+	logrus.Debugf("Pushing metadata to registry at %s", b.src)
+	return b.pushImage(ctx, data, fpath)
 }
 
-// GetWriter returns an os.File as a writer.
+// GetWriter returns an os.File as a writeb.
 // In this implementation, key is a file path.
-func (r *registryBackend) GetWriter(ctx context.Context, fpath string) (io.Writer, error) {
-	return r.localDirBackend.GetWriter(ctx, fpath)
+func (b *registryBackend) GetWriter(ctx context.Context, fpath string) (io.Writer, error) {
+	return b.localDirBackend.GetWriter(ctx, fpath)
+}
+
+// Open reads the provided object from a registry source and provides an io.ReadCloser
+func (b *registryBackend) Open(fpath string) (io.ReadCloser, error) {
+	// Assumes this is in place on disk
+	// QUESTION(jpower): can we make this assumption here?
```

Comment:
The returned `io.ReadCloser`'s `Close()` should push the image containing metadata.